### PR TITLE
speeds up will_fusion_create_cycle

### PR DESCRIPTION
**Pull request by ngimel**
improves #102622 from ~150s to ~15s. 
The way computing recursive predecessors works is if `nodes = node1.recursive_predecessors` then recursive predecessors of any `n` in `nodes` should still be a subset of `nodes`, so we can shortcut computing intersection of `node.recursive_predecessors - combined_predecessors`

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225

### CODE REVIEW

File: torch/_inductor/scheduler.py [hunk 1]
```diff
@@ -1027,10 +1027,21 @@ def will_fusion_create_cycle(self, node1, node2):
...
-                    check(self.name_to_fused_node[n])
-                    for n in node.recursive_predecessors - combined_predecessors
-                )
+                cond0 = bool(combined_names & node.recursive_predecessors)
+
+                if cond0:
+                    return cond0
+
+                names = node.get_names()
+                shortcut = all(n in combined_predecessors for n in names)
```
Comment (Chillee) [inline torch/_inductor/scheduler.py:1036]: I'm not sure I see why this is equivalent 🤔

Comment (ngimel) [reply torch/_inductor/scheduler.py:1036]: If all the fused node outputs are in `combined_predecessors`, then all the `recursive_predecessors` of the node will also be in `combined_predecessors` and thus the intersection will be empty. If `node` actually has something else then we go on the old path.

Comment (lezcano) [reply torch/_inductor/scheduler.py:1036]: Not sure about the general logic, but this line is equivalent to
```suggestion
                shortcut = name.issubset(combined_predecessors)
```

**Review by Chillee — APPROVED** [review]
Talked with Natalia, I am convinced this is correct.

**Pull request closed without merging**